### ISSUE TITLE
Need to add `dolfinx.mesh.MeshTagsMetaClass.mesh` to python layer

### ISSUE THREAD

**Issue by francesco-ballarin** [FEniCS/dolfinx#2519]
The following snippet
```
import dolfinx.mesh
import mpi4py.MPI
import numpy as np

mesh = dolfinx.mesh.create_unit_square(mpi4py.MPI.COMM_WORLD, 2, 2)
cell_entities = dolfinx.mesh.locate_entities(
    mesh, mesh.topology.dim, lambda x: np.full((x.shape[1], ), True))
cell_tags = dolfinx.mesh.meshtags(
    mesh, mesh.topology.dim, cell_entities,
    np.full(cell_entities.shape, fill_value=1, dtype=np.int32))
print(type(cell_tags.mesh), type(mesh))
```
produces
```
<class 'dolfinx.cpp.mesh.Mesh'> <class 'dolfinx.mesh.Mesh'>
```
after #2500, instead of
```
<class 'dolfinx.mesh.Mesh'> <class 'dolfinx.mesh.Mesh'>
```

I believe that `dolfinx.mesh.MeshTagsMetaClass.mesh` should be added to the python layer, wrapping the C++ output in a call to `Mesh.from_cpp`. Unfortunately I can't come up with a PR myself because I am not sure what would be the UFL domain argument to be passed to that call :(

**Comment by garth-wells** [comment]
Yes, we need to fix this. I came across another issue with meshtags not having the right int type for an assembler giving a very cryptic message.

The UFL domain argument is a pain, and will hopefully get fixed in a new UFL. Would be better if we had requirement on what the mesh can *do* rather than *what it is*.

Despite having added the `FooMetaClass` classes myself, I'm not sure it's the best solution. Maybe it's simpler to to just add a `dolfinx.mesh.MeshTags` class, just as is now done for `dolfinx.mesh.Mesh`.

**Comment by jorgensd** [comment]
> Despite having added the `FooMetaClass` classes myself, I'm not sure it's the best solution. Maybe it's simpler to to just add a `dolfinx.mesh.MeshTags` class, just as is now done for `dolfinx.mesh.Mesh`.

I would be for using the same layout for MeshTags and Form as we do for Mesh.
